uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestCalendarFeedConfigTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequestCalendarFeedConfig as LeaveRequestCalendarFeedConfig;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_BAO_LeaveRequestCalendarFeedConfigTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_BAO_LeaveRequestCalendarFeedConfigTest extends BaseHeadlessTest {
+
+  public function testDefaultParametersAreSetWhenCreatingACalendarConfig() {
+    $leaveFeedConfig = LeaveRequestCalendarFeedConfig::create([
+      'title' => 'Feed 1',
+      'timezone' => 'America/Monterrey',
+    ]);
+
+    $this->assertNotNull($leaveFeedConfig->hash);
+    $dateNow = new DateTime();
+    $this->assertEquals($dateNow, new DateTime($leaveFeedConfig->created_date), '', 10);
+  }
+
+  public function testDefaultParametersCanNotBeManipulatedWhenCreatingACalendarFeedConfig() {
+    $hash = '5aejfkfdjJJU';
+    $createdDate = CRM_Utils_Date::processDate('2016-01-01');
+
+    $leaveFeedConfig = LeaveRequestCalendarFeedConfig::create([
+      'title' => 'Feed 1',
+      'timezone' => 'America/Monterrey',
+      'hash' => $hash,
+      'created_date' => $createdDate
+    ]);
+
+    $this->assertNotEquals($leaveFeedConfig->hash, $hash);
+    $this->assertNotEquals($leaveFeedConfig->created_date, $createdDate);
+  }
+
+  public function testDefaultParametersCanNotBeManipulatedWhenUpdatingACalendarFeedConfig() {
+    $hash = '5aejfkfdjJJU';
+    $createdDate = CRM_Utils_Date::processDate('2016-01-01');
+
+    $leaveFeedConfig1 = LeaveRequestCalendarFeedConfig::create([
+      'title' => 'Feed 1',
+      'timezone' => 'America/Monterrey',
+    ]);
+
+    $leaveFeedConfig = LeaveRequestCalendarFeedConfig::create([
+      'id' => $leaveFeedConfig1->id,
+      'hash' => $hash,
+      'created_date' => $createdDate
+    ]);
+
+    $this->assertNotEquals($leaveFeedConfig->hash, $hash);
+    $this->assertNotEquals($leaveFeedConfig->created_date, $createdDate);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestCalendarFeedConfigException
+   * @expectedExceptionMessage Please add a valid timezone for the leave request calendar feed configuration
+   */
+  public function testCreateWillThrowAnExceptionWhenTimezoneIsNotValid() {
+    LeaveRequestCalendarFeedConfig::create([
+      'title' => 'Feed 1',
+      'timezone' => 'America/Whatever',
+    ]);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestCalendarFeedConfigException
+   * @expectedExceptionMessage A leave request calendar feed configuration with same title already exists!
+   */
+  public function testCreateWillThrowAnExceptionWhenTitleAlreadyExists() {
+    LeaveRequestCalendarFeedConfig::create([
+      'title' => 'Feed 1',
+      'timezone' => 'Europe/Stockholm',
+    ]);
+
+    LeaveRequestCalendarFeedConfig::create([
```

Comment:
For this test, I think that using `setExpectedException()` and `setExpectedExceptionMessage()` are better options than the annotations. With the methods you can make sure the test will only pass if the second `create()` call is the one that throws the exception